mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAOptimizer.scala

```diff
@@ -503,17 +518,15 @@ final class OnlineLDAOptimizer extends LDAOptimizer {
   }
 
   /**
-   * Update alpha based on `gammat`, the inferred topic distributions for documents in the
-   * current mini-batch. Uses Newton-Rhapson method.
+   * Update alpha based on `logphat`.
```

Comment:
I also rename `N` to `batchSize` which it is.

Reply:
There're a small difference between old `N` and `batchSize`. `N` in old version code do not count non-empty docs.

Reply:
@WeichenXu123, you are right. So should we add `stats.count()` or should we rather embed the counting in the aggregation phase so that we avoid the second pass?

Reply:
Or another suggestion. Lets, have smth like 
`val stats: RDD[(BDM[Double], Option[BDV[Double]], Int)] = batch.mapPartitions {...}`  where the `Int` stands for the number of non-empty elements in a partition.  

What do you think?

Reply:
I think it's OK. Let's keep the logic exactly the same with old version.

Reply:
But should we have `val stats: RDD[(BDM[Double], Option[BDV[Double]], Int)] = batch.mapPartitions {...}` stuff?

Reply:
Please, check out the updated PR. 

I have added `val stats: RDD[(BDM[Double], Option[BDV[Double]], Int)] = batch.mapPartitions {...}`.
Unfortunately, we cannot have the aggregation operation in a purely in-place manner now since `Int` is immutable. Shouldn't be a big deal since matrices and vectors are still updated in place.

Reply:
I think it doesn't matter if `seq` do not return left param reference but return a new object.